src/main.rs

```diff
@@ -411,16 +429,22 @@ async fn attest(
         })
         .collect::<Result<Vec<_>, _>>()
         .and_then(|issued| {
-            let issued = issued
+            let issued: Vec<Certificate<'_>> = issued
                 .iter()
                 .map(|c| Certificate::from_der(c).or(Err(StatusCode::INTERNAL_SERVER_ERROR)))
                 .collect::<Result<_, _>>()?;
-            Output {
-                chain: vec![issuer],
-                issued,
-            }
-            .to_vec()
-            .or(Err(StatusCode::INTERNAL_SERVER_ERROR))
+
+            let response = if one_cert {
+                vec![issuer, issued[0].clone()].to_vec()
+            } else {
+                Output {
+                    chain: vec![issuer],
+                    issued,
+                }
+                .to_vec()
+            };
+
+            response.or(Err(StatusCode::INTERNAL_SERVER_ERROR))
```

Comment:
Either
```suggestion
            if one_cert {
                vec![issuer, issued[0].clone()].to_vec()
            } else {
                Output {
                    chain: vec![issuer],
                    issued,
                }
                .to_vec()
            }
            .or(Err(StatusCode::INTERNAL_SERVER_ERROR))
```

Or, even better, handle each error separately. Ideally, we'd want to add some logs/more descriptive errors here anyway and we need to differentiate between the two cases

Reply:
Improved as per this recommendation. It's definitely cleaner and I had thought about it, but wasn't sure.